### PR TITLE
Add l2_kar_options option for L2 guest

### PR DESCRIPTION
**Pull request by qcheng-redhat**
ID: 1816458

Add  l2_kar_options option for L2 guest

Signed-off-by: Qinghua Cheng <qcheng@redhat.com>

### CODE REVIEW

**Comment by qcheng-redhat** [comment]
@CongLi @luckyh @yanan-fu Please help review. Thanks.

**Comment by qcheng-redhat** [comment]
Test results:
# python3 ./ConfigTest.py --testcase=nested_test --guestname=RHEL.8.2.0 --customsparams="l2_guest_name=RHEL.8.2.0\nmachine_type=q35\nauto_cpu_model=no\ncpu_model=host" --clone=no --machine=q35
 (1/1) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.nested_test.boot_l2.q35: PASS (154.62 s)

# python3 ./ConfigTest.py --testcase=nested_test --guestname=RHEL.8.2.0 --customsparams="l2_guest_name=RHEL.8.2.0\nauto_cpu_model=no\ncpu_model=host" --clone=no
 (1/1) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.nested_test.boot_l2: PASS (141.39 s)

File: qemu/tests/cfg/nested_test.cfg [hunk 1]
```diff
@@ -2,6 +2,7 @@
     kar_repo = <kar_repo_url>
     cert_url = <cert_ca_url>
     nested_bs_options = ""
+    machine_type = ""
```
Comment (yanan-fu) [inline qemu/tests/cfg/nested_test.cfg]: `machine_type` is a param that already be used by L1, need use different param name for L2
such as `l2_machine_type`

Comment (qcheng-redhat) [reply qemu/tests/cfg/nested_test.cfg]: Fixed.

**Comment by qcheng-redhat** [comment]
python3 ./ConfigTest.py --testcase=nested_test --guestname=RHEL.8.2.0 --customsparams="l2_guest_name=RHEL.8.2.0\nl2_machine_type=q35\nauto_cpu_model=no\ncpu_model=host" --clone=no --machine=q35
 (1/1) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.nested_test.boot_l2.q35: PASS (149.21 s)

python3 ./ConfigTest.py --testcase=nested_test --guestname=RHEL.8.2.0 --customsparams="l2_guest_name=RHEL.8.2.0\nauto_cpu_model=no\ncpu_model=host" --clone=no
 (1/1) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.nested_test.boot_l2: PASS (157.04 s)

**Comment by yanan-fu** [comment]
@qingwangrh  For rhel8, we test `q35` with high priority, do we need to set `q35` as the default machine type for l2 guest ?

**Comment by qcheng-redhat** [comment]
I just would like to make the default value same for l1 and l2, so did not set default value as q35 for l2 guest.

**Comment by yanan-fu** [comment]
@qcheng-redhat  I wonder if there are other kar options for L2 guest for this feature in the future, such as driveformat or something else.

If no, current change is ok for me.
If there are,  how do you think about use one param to represent all ? In this way, the vlaue of the new param include option name and value, such as:
```
l2_kar_options = '--machine=q35 --driveformat=**'
kar_cmd += "  %s" % l2_kar_options
```

**Comment by qcheng-redhat** [comment]
Good suggestion. Fixed. And here are results:
python3 ./ConfigTest.py --testcase=nested_test --guestname=RHEL.8.2.0 --customsparams="l2_guest_name=RHEL.8.2.0\nl2_kar_options=--machine=q35 --firmware=ovmf\nauto_cpu_model=no\ncpu_model=host" --clone=no --machine=q35
(1/1) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.nested_test.boot_l2.q35: PASS (1681.27 s)

python3 ./ConfigTest.py --testcase=nested_test --guestname=RHEL.8.2.0 --customsparams="l2_guest_name=RHEL.8.2.0\nl2_kar_options=--machine=q35\nauto_cpu_model=no\ncpu_model=host" --clone=no --machine=q35
 (1/1) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.nested_test.boot_l2.q35: PASS (145.45 s)

**Comment by yanan-fu** [comment]
Acked-by: Yanan Fu <yfu@redhat.com>

@luckyh  Do you have any other comment ?

**Comment by yanan-fu** [comment]
@qcheng-redhat please update the commit info, then let's merge it

**Comment by qcheng-redhat** [comment]
Fixed.

**Pull request merged**